libs/pmp-api/api-service/repository/data-access/src/lib/repositories/pr.data-service.ts

```diff
@@ -0,0 +1,56 @@
+import { HttpService, Injectable } from '@nestjs/common';
+import {
+  githubConfig,
+  PmpApiServiceConfigService
+} from '@pimp-my-pr/pmp-api/shared/config';
+import { urlFactory } from '@valueadd/typed-urls';
+import { map } from 'rxjs/operators';
+import { AxiosResponse } from 'axios';
+import { PrChanges } from '@pimp-my-pr/pmp-api/api-service/repository/domain';
+import { GithubPrFilesEntity } from '../domain/entities/github-pr-files.entity';
+import { catchRequestExceptions } from '@pimp-my-pr/pmp-api/shared/util';
+
+@Injectable()
+export class PrDataService {
+  endpoints = {
+    getPrFiles: urlFactory<'repoFullName' | 'prId'>(
+      githubConfig.apiUrl + '/repos/:repoFullName/pulls/:prId/files',
+      true
+    )
+  };
+
+  constructor(
+    private httpService: HttpService,
+    private pmpApiServiceConfigService: PmpApiServiceConfigService
+  ) {}
+
+  getPrChanges(repoFullName: string, prId: number): Promise<PrChanges> {
+    return this.httpService
+      .get<GithubPrFilesEntity[]>(
+        this.endpoints.getPrFiles.url({
+          repoFullName,
+          prId
+        }),
+        {
+          headers: {
+            Authorization:
+              'token ' + this.pmpApiServiceConfigService.getGithubToken()
```

Comment:
Maybe it should be in some interceptor due to the fact that it's already duplicated two times?